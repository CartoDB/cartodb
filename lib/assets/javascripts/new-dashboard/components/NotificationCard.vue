<template>
<div class="notification">
  <div class="notification-info">
    <div class="notification-received text is-small">{{ receivedAtFormatted }}</div>
    <span v-if="!readAt" class="notification-read"></span>
  </div>
  <div v-html="htmlBody" class="notification-html text is-caption" :class="{'is-unread': !readAt}"></div>
</div>
</template>

<script>
import format from 'date-fns/format';
export default {
  name: 'NotificationCard',
  props: {
    id: String,
    receivedAt: String,
    readAt: String,
    htmlBody: String
  },
  computed: {
    receivedAtFormatted () {
      return format(new Date(this.receivedAt), 'DD/MM/YYYY');
    }
  }
};
</script>
<style scoped lang="scss">
@import 'stylesheets/new-dashboard/variables';

.notification-info {
  display: flex;
  align-items: center;
  margin-bottom: 16px;
}

.notification-received {
  color: $text-secondary-color;
}

.notification-read {
  display: block;
  width: 12px;
  height: 12px;
  margin-left: 24px;
  border-radius: 50%;
  background: $notification;
}

.notification-html {
  margin-top: 16px;
  color: $text-color;

  &.is-unread {
    font-weight: 600;
  }
}

</style>
