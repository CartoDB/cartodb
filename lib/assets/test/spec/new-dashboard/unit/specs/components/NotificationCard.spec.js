import {
  mount
} from '@vue/test-utils';
import NotificationCard from 'new-dashboard/components/NotificationCard';
import mockNotifications from '../fixtures/notifications';

describe('NotificationCard.vue', () => {
  let notifications, notificationCardWrapper;
  beforeEach(() => {
    notifications = mockNotifications;
    notificationCardWrapper = mount(NotificationCard);
  });

  describe('basic behaviour', () => {
    it('should render the body', () => {
      notificationCardWrapper.setProps({
        htmlBody: notifications[0].html_body
      });
      expect(notificationCardWrapper).toMatchSnapshot();
    });

    it('should render the date properly formatted', () => {
      notificationCardWrapper.setProps({
        receivedAt: notifications[0].received_at
      });

      expect(notificationCardWrapper).toMatchSnapshot();
    });

    it('should hide the green dot when the notification has been read', () => {
      notificationCardWrapper.setProps({
        readAt: notifications[1].read_at
      });

      expect(notificationCardWrapper).toMatchSnapshot();
    });

    it('should show the green dot dot when the notification has not been read', () => {
      notificationCardWrapper.setProps({
        readAt: notifications[0].read_at
      });

      expect(notificationCardWrapper).toMatchSnapshot();
    });
  });
});
