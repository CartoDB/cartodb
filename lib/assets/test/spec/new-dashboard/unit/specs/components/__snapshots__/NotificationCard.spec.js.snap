// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationCard.vue basic behaviour should hide the green dot when the notification has been read 1`] = `
<div class="notification">
  <div class="notification-info">
    <div class="notification-received text is-small">Invalid Date</div>
    <!---->
  </div>
  <div class="notification-html text is-caption"></div>
</div>
`;

exports[`NotificationCard.vue basic behaviour should render the body 1`] = `
<div class="notification">
  <div class="notification-info">
    <div class="notification-received text is-small">Invalid Date</div> <span class="notification-read"></span>
  </div>
  <div class="notification-html text is-caption is-unread">
    <p>Im an unread notification, I have bold text and a green dot on top.</p>
  </div>
</div>
`;

exports[`NotificationCard.vue basic behaviour should render the date properly formatted 1`] = `
<div class="notification">
  <div class="notification-info">
    <div class="notification-received text is-small">20/11/2018</div> <span class="notification-read"></span>
  </div>
  <div class="notification-html text is-caption is-unread"></div>
</div>
`;

exports[`NotificationCard.vue basic behaviour should show the green dot dot when the notification has not been read 1`] = `
<div class="notification">
  <div class="notification-info">
    <div class="notification-received text is-small">Invalid Date</div> <span class="notification-read"></span>
  </div>
  <div class="notification-html text is-caption is-unread"></div>
</div>
`;
